fdbserver/include/fdbserver/IPageEncryptionKeyProvider.actor.h

```diff
@@ -68,12 +69,8 @@ class IPageEncryptionKeyProvider : public ReferenceCounted<IPageEncryptionKeyPro
 	// Expected encoding type being used with the encryption key provider.
 	virtual EncodingType expectedEncodingType() const = 0;
 
-	// Checks whether encryption should be enabled. If not, the encryption key provider will not be used by
-	// the pager, and instead the default non-encrypted encoding type (XXHash64) is used.
-	virtual bool enableEncryption() const = 0;
-
-	// Whether encryption domain is enabled.
-	virtual bool enableEncryptionDomain() const { return false; }
+	// Whether domain-aware encryption is enabled.
+	virtual bool isDomainAware() const { return false; }
```

Comment:
do we need this method? I mean it can be inferred once we have `db.encryptionAtRestMode`?

Reply:
Also, making it `virtual` allows inherited class to override the implementation, which doesn't feel right.

Reply:
I think I'm creating more confusion by renaming the method from `enableEncryptionDomain`. It is different to `encryptionAtRestMode`, in that `AESEncryptionKeyProvider` always has `isDomainAware()=true` in both `cluster-aware` and `domain-aware` mode, where it needs to have the system keys in its own domain. I think I shall rename it back to `enableEncryptionDomain`. Does it make sense?